File: morphac/constants/python/test/environment_constants_test.py
import pytest

from morphac.constants.environment_constants import MapConstants


def test_map_constants():

    assert MapConstants.EMPTY == 0
    assert MapConstants.OBSTACLE == 1


def test_set_map_constants():

    # The values should not be modifiable.

    with pytest.raises(AttributeError):
        MapConstants.EMPTY = 0
    with pytest.raises(AttributeError):
        MapConstants.OBSTACLE = 1

File: morphac/constants/test/environment_constants_test.cc
#include "constants/include/environment_constants.h"

#include "gtest/gtest.h"

namespace {

using morphac::constants::MapConstants;

class EnvironmentConstantsTest : public ::testing::Test {
 protected:
  EnvironmentConstantsTest() {}

  void SetUp() override {}
};

TEST_F(EnvironmentConstantsTest, MapConstants) {
  // Test Map constants.
  ASSERT_EQ(MapConstants::EMPTY, 0);
  ASSERT_EQ(MapConstants::OBSTACLE, 1);
}

}  // namespace

int main(int argc, char** argv) {
  ::testing::InitGoogleTest(&argc, argv);
  return RUN_ALL_TESTS();
}
